### ISSUE TITLE
Does not work with django-auto-prefetch models and fields

### ISSUE THREAD

**Issue by yatharth** [typeddjango/django-stubs#1467]
# Bug report

## What's wrong

`django-stubs` fails to work with models using [`django-auto-prefetch`](https://github.com/tolomea/django-auto-prefetch).

If I have a class like this:

```
import 

class Bar(auto_prefetch.Model):
   bar = CharField(max_length=10)

class Foo(auto_prefetch.Model):
    foo = auto_prefetch.ForeignKey(
        Bar, on_delete=models.PROTECT
    )

```

and then I try to typecheck this code

```
def hello(foo: Foo):
    print(foo.bar)
```

I get errors like this:

```
[...].py:[...]: error: "Model" has no attribute "bar"
```

## How is that should be

Whatever `django-stubs` is doing to make regular models and ForeignKeys work should also work with models that are inheriting from `auto_prefetch.Model`.

## System information

- OS: macOS 13.1
- `python` version: 3.11
- `django` version: 3.2.18
- `mypy` version: 0.961
- `django-stubs` version: 1.12.0
- `django-stubs-ext` version: 0.5.0

**Comment by sobolevn** [comment]
Just a tip, that you can lie about the base class with `if TYPE_CHECKING: _base: TypeAlias = Model; else: _base: TypeAlias = auto_prefetch.Model`

**Comment by flaeppe** [comment]
I'm not sure this is an issue with django-stubs? Since `django-auto-prefetch` doesn't seem to include package data for `py.typed`, essentially making it an untyped package?

Unless I'm missing something

**Comment by flaeppe** [comment]
I've also tried inherited `models.Model` in the past, which has gotten stuff working for me:

```python
class Foo(auto_prefetch.Model, models.Model):
    foo = auto_prefetch.ForeignKey(
        Bar, on_delete=models.PROTECT
    )
```

**Comment by intgr** [comment]
Actually it *does* have `py.typed`, courtesy of our own adamchainz 😄

https://github.com/tolomea/django-auto-prefetch/tree/main/src/auto_prefetch

Maybe our detection of `Model` classes doesn't consider subclasses of Django's `Model`?

**Comment by flaeppe** [comment]
It has a `py.typed`. But it's not included in `package_data` in `setup.cfg` (I haven't inspected a pypi installation though, just browsing around and comparing config to other packages)

**Comment by flaeppe** [comment]
I just had a look at a fresh pypi installation though, it looks fine. So I was wrong 🙂 


```
 ➤ pip show -f django-auto-prefetch
Name: django-auto-prefetch
Version: 1.5.1
Summary: Automatically prefetch foreign key values as needed.
Home-page: https://github.com/tolomea/django-auto-prefetch
Author: Gordon Wrigley
Author-email: gordon.wrigley@gmail.com
License: BSD-3-Clause
Location: ./.venv/lib/python3.10/site-packages
Requires: Django
Required-by:
Files:
  auto_prefetch/__init__.py
  auto_prefetch/__pycache__/__init__.cpython-310.pyc
  auto_prefetch/py.typed
  django_auto_prefetch-1.5.1.dist-info/INSTALLER
  django_auto_prefetch-1.5.1.dist-info/LICENSE
  django_auto_prefetch-1.5.1.dist-info/METADATA
  django_auto_prefetch-1.5.1.dist-info/RECORD
  django_auto_prefetch-1.5.1.dist-info/REQUESTED
  django_auto_prefetch-1.5.1.dist-info/WHEEL
  django_auto_prefetch-1.5.1.dist-info/top_level.txt
```

**Comment by adamchainz** [comment]
Yes, the package has basic typing. I haven't sorted out making it strict-mode-compatible yet (https://github.com/tolomea/django-auto-prefetch/pull/217) so perhaps some missing generic parameters would help there. If you have time to try out on that PR, it would be useful.
